### ISSUE TITLE
future incompatible dependencies

### ISSUE THREAD

**Issue by Nanai10a** [zellij-org/zellij#2295]
I did `cargo install zellij`, and seen:

```
warning: the following packages contain code that will be rejected by a future version of Rust: daemonize v0.4.1, nom v5.1.2
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 1`
```

with using:

```
rustup 1.25.2 (2023-02-04)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
info: The currently active `rustc` version is `rustc 1.70.0-nightly (171693274 2023-03-14)`
```

here is a dependency tree:

- zellij v0.36.0
  - zellij-server v0.36.0
    - **daemonize v0.4.1**
  - zellij-client v0.36.0 
    - zellij-utils v0.36.0 
      - termwiz v0.20.0
        - terminfo v0.7.3
          - **nom v5.1.2**

and report:

```
To solve this problem, you can try the following approaches:


- Some affected dependencies have newer versions available.
You may want to consider updating them to a newer version to see if the issue has been fixed.

daemonize v0.4.1 has the following newer versions available: 0.5.0
nom v5.1.2 has the following newer versions available: 6.0.0-alpha1, 6.0.0-alpha2, 6.0.0-alpha3, 6.0.0-beta1, 6.0.0-beta2, 6.0.0-beta3, 6.0.0-beta4, 6.0.0-beta5, 6.0.0, 6.0.1, 6.1.0, 6.1.1, 6.1.2, 6.2.0, 6.2.1, 7.0.0-alpha1, 7.0.0-alpha2, 7.0.0-alpha3, 7.0.0, 7.1.0, 7.1.1, 7.1.2, 7.1.3
terminfo v0.7.3 has the following newer versions available: 0.7.4, 0.7.5


- If the issue is not solved by updating the dependencies, a fix has to be
implemented by those dependencies. You can help with that by notifying the
maintainers of this problem (e.g. by creating a bug report) or by proposing a
fix to the maintainers (e.g. by creating a pull request):

  - daemonize@0.4.1
  - Repository: https://github.com/knsd/daemonize
  - Detailed warning command: `cargo report future-incompatibilities --id 3 --package daemonize@0.4.1`

  - nom@5.1.2
  - Repository: https://github.com/Geal/nom
  - Detailed warning command: `cargo report future-incompatibilities --id 3 --package nom@5.1.2`

  - terminfo@0.7.3
  - Repository: https://github.com/meh/rust-terminfo
  - Detailed warning command: `cargo report future-incompatibilities --id 3 --package terminfo@0.7.3`

- If waiting for an upstream fix is not an option, you can use the `[patch]`
section in `Cargo.toml` to use your own version of the dependency. For more
information, see:
https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#the-patch-section
```

Thanks.

**Comment by har7an** [comment]
Hi,

thanks for the report! I had a look at this and updated `daemonize`, but at the moment there's nothing we can do about the other package: `terminfo@0.7.3` is a dependency of `termwiz@0.20.0`, which doesn't have a more recent version available. The development version has updated to `termwiz@0.8` though, so the problem will resolve itself in the future. Besides, we're probably still a long way off from using rust 1.70 - In any case: Thank you very much for reporting this!

This will be partially resolved as part of #2375.

**Comment by mati865** [comment]
This issue is not fixed as `zellij-server` still uses `daemonize = "0.4.1"`.
